packages/react-reconciler/src/forks/ReactFiberHostConfig.bun.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export * from 'react-dom-bindings/src/client/ReactDOMHostConfig';
```

Comment:
I'm not sure if you've already explored this and found it was required (I'm not terribly familiar with all the nuances of the forking implementation we have) but I am surprised the HostConfig needs a bun fork since this code doesn't ever run in the Bun runtime.

Presumably it is because the react-dom/client bundle it part of `BUN_DEV` and `BUN_PROD`? I am guessing there is a way to not need

cc @sebmarkbage

Reply:
I'm not an expert either, but it seems that once you go down the path of adding a new fork, you need to shim everything for Flow to be happy. This is because the generated Flow configs remap a bunch of imports unconditionally: https://github.com/colinhacks/react/blob/add-bun-fizz-server/scripts/flow/createFlowConfigs.js#L56-L61

Interestingly I can `return null` from the `forks.js` resolver for `react-reconciler/src/ReactFiberHostConfig.js` and the generated bundle doesn't change, so you're correct that it isn't being bundled. But I think this is the path of least resistance until the forking gets a little more clever (or doesn't - a few unnecessary re-exports never hurt anyone 😛)